sns_ik_lib/test/sns_acc_ik_base_test.cpp

```diff
@@ -0,0 +1,151 @@
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <gtest/gtest.h>
+
+#include <Eigen/Dense>
+#include <ros/console.h>
+
+#include <sns_ik/sns_acc_ik_base.hpp>
+#include "rng_utilities.hpp"
+
+/*************************************************************************************************/
+
+void checkEqualVector(const Eigen::VectorXd& A, const Eigen::VectorXd& B, double tol)
+{
+  ASSERT_EQ(A.size(), B.size());
+  int n = A.size();
+  for (int i = 0; i < n; i++) {
+    ASSERT_NEAR(A(i), B(i), tol);
+  }
+}
+
+/*************************************************************************************************/
+
+void checkVectorLimits(const Eigen::VectorXd& low, const Eigen::VectorXd& val, const Eigen::VectorXd& upp, double tol)
+{
+  ASSERT_EQ(low.size(), val.size());
+  ASSERT_EQ(upp.size(), val.size());
+  int n = val.size();
+  for (int i = 0; i < n; i++) {
+    ASSERT_LE(val(i), upp(i) + tol);
+    ASSERT_GE(val(i), low(i) - tol);
+  }
+}
+
+/*************************************************************************************************/
+
+/*
+ * This test is for the solveAccIkBasic()
+ */
+TEST(sns_acc_ik_base, basic_no_limits)
+{
+  sns_ik::rng_util::setRngSeed(75716, 11487);  // set the initial seed for the random number generators
+  int nTest = 100;
+  double tol = 1e-10;
+  for (int iTest = 0; iTest < nTest; iTest++) {
+
+    // generate a test problem
+    int nTask = sns_ik::rng_util::getRngInt(0, 1, 6);
+    int nJoint = sns_ik::rng_util::getRngInt(0, nTask, nTask + 4);
+    Eigen::MatrixXd J = sns_ik::rng_util::getRngMatrixXd(0, nTask, nJoint, -1.0, 1.0);
+    Eigen::MatrixXd dJdq = sns_ik::rng_util::getRngVectorXd(0, nTask, -1.0, 1.0);
+    Eigen::VectorXd ddx = sns_ik::rng_util::getRngVectorXd(0, nTask, -1.0, 1.0);
+    Eigen::VectorXd tolVec = tol * Eigen::VectorXd::Ones(nJoint);
+    Eigen::VectorXd ddq;
+    double taskScale;
+
+    // solve
+    sns_ik::SnsAccIkBase::uPtr ikSolver = sns_ik::SnsAccIkBase::create(nJoint);
+    ASSERT_TRUE(ikSolver.get() != nullptr);
+    sns_ik::SnsAccIkBase::ExitCode exitCode = ikSolver->solve(J, dJdq, ddx, &ddq, &taskScale);
+    ASSERT_TRUE(exitCode == sns_ik::SnsAccIkBase::ExitCode::Success);
+
+    // check requirements
+    ASSERT_LE(taskScale, 1.0 + tol);
+    ASSERT_GT(taskScale, 0.0);
+    checkEqualVector(taskScale * ddx, J * ddq + dJdq, tol);
+  }
+}
+
+/*************************************************************************************************/
+
+/*
+ * This test is for the solveAccIkBasic()
```

Comment:
Can you add some more description here?  I think the function name in the existing documentation might also be wrong.